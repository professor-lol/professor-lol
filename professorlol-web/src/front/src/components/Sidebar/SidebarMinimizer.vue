<template>
  <button class="sidebar-minimizer" type="button" @click="onClick()"></button>
</template>
<script>
import { togglePs } from '../../mixins/togglePs'

export default {
  name: 'sidebar-minimizer',
  mixins: [ togglePs ],
  mounted: function () {
    const isMinimized = document.body.classList.contains('sidebar-minimized')
    this.togglePs(!isMinimized)
  },
  methods: {
    onClick () {
      this.sidebarMinimize()
      this.brandMinimize()
    },
    sidebarMinimize () {
      const isMinimized = document.body.classList.toggle('sidebar-minimized')
      this.$emit('cui-sidebar-minimize', isMinimized)
      this.togglePs(!isMinimized)
    },
    brandMinimize () {
      document.body.classList.toggle('brand-minimized')
    }
  }
}
</script>
