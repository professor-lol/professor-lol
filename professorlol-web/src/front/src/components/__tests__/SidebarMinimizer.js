import { mount } from "@vue/test-utils";
import SidebarMinimizer from "../Sidebar/SidebarMinimizer";

describe("SidebarMinimizer.vue", () => {
  // Inspect the raw component options
  it('has onClick method', () => {
    expect(typeof SidebarMinimizer.methods.onClick).toBe('function')
  })
  it('has sidebarMinimize method', () => {
    expect(typeof SidebarMinimizer.methods.sidebarMinimize).toBe('function')
  })
  it('has brandMinimize method', () => {
    expect(typeof SidebarMinimizer.methods.brandMinimize).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(SidebarMinimizer)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.classes()).toContain('sidebar-minimizer')
  })
});
