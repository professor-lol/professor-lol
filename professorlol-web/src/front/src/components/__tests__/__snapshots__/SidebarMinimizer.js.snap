// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMinimizer.vue renders correctly 1`] = `
<button
  class="sidebar-minimizer"
  type="button"
/>
`;
